src/Logging/test/DynamicSerilog.Test/SerilogWebHostBuilderTest.cs

```diff
@@ -13,84 +15,82 @@
 
         _ = new WebHostBuilder().UseStartup<Startup>().AddDynamicSerilog((_, loggerConfiguration) =>
-        {
             loggerConfiguration.MinimumLevel.Error().Enrich.WithExceptionDetails().MinimumLevel.Override("Microsoft", LogEventLevel.Warning).WriteTo
-                .Sink(testSink);
-        }).Build();
-
-        await Task.Delay(1000);
+                .Sink(testSink)).Build();
 
         List<string> logs = testSink.GetLogs();
 
-        Assert.NotEmpty(logs);
-        Assert.Contains("error", logs);
-        Assert.DoesNotContain("info", logs);
+        logs.Should().NotBeEmpty();
+        logs.Should().Contain("error");
+        logs.Should().NotContain("info");
     }
 
     [Fact]
     public void OnlyApplicableFilters_AreApplied_via_Options()
     {
-        var appsettings = new Dictionary<string, string>
+        var appSettings = new Dictionary<string, string?>
         {
             { "Serilog:Using:0", "Steeltoe.Logging.DynamicSerilog.Test" },
             { "Serilog:MinimumLevel:Default", "Error" },
             { "Serilog:MinimumLevel:Override:Microsoft", "Warning" },
             { "Serilog:WriteTo:Name", "TestSink" }
         };
 
-        IWebHost host = new WebHostBuilder().ConfigureAppConfiguration(builder => builder.AddInMemoryCollection(appsettings)).UseStartup<Startup>()
+        IWebHost host = new WebHostBuilder().ConfigureAppConfiguration(builder => builder.AddInMemoryCollection(appSettings)).UseStartup<Startup>()
             .AddDynamicSerilog().Build();
 
-        object logger = host.Services.GetService(typeof(ILogger<SerilogDynamicWebHostBuilderTest>));
+        object logger = host.Services.GetRequiredService<ILogger<SerilogWebHostBuilderTest>>();
         ILogEventSink[] sinks = GetSinks(logger);
-        Assert.NotNull(sinks);
-        TestSink testSink = sinks.OfType<TestSink>().FirstOrDefault();
-        Assert.NotNull(testSink);
+        sinks.Should().NotBeEmpty();
 
-        List<string> logs = testSink.GetLogs();
-        Assert.NotEmpty(logs);
-        Assert.Contains("error", logs);
-        Assert.DoesNotContain("info", logs);
+        TestSink? testSink = sinks.OfType<TestSink>().FirstOrDefault();
+        testSink.Should().NotBeNull();
+
+        List<string> logs = testSink!.GetLogs();
+
+        logs.Should().NotBeEmpty();
+        logs.Should().Contain("error");
+        logs.Should().NotContain("info");
     }
 
-    public static ILogEventSink[] GetSinks(object logger)
+    internal static ILogEventSink[] GetSinks(object logger)
     {
-        FieldInfo loggerField = logger.GetType().GetField("_logger", BindingFlags.NonPublic | BindingFlags.Instance);
-        object logger2 = loggerField.GetValue(logger);
-        PropertyInfo loggersField = logger2.GetType().GetProperty("Loggers");
-        var loggersValueArray = loggersField.GetValue(logger2) as Array;
+        FieldInfo loggerField = logger.GetType().GetField("_logger", BindingFlags.NonPublic | BindingFlags.Instance)!;
```

Comment:
Not a showstopper, but I'm wondering if we have something in Steeltoe.Common that would clean up some of this reflection?

Reply:
I tried replacing that with shims, but it got messy so I reverted that. The code has deep assumptions about the internal structure of objects and array elements of various types, so wrapping that wouldn't be generally reusable.